### ISSUE TITLE
Can we get a wiki page or something for how to get a minimum working start project?

### ISSUE THREAD

**Issue by se5a** [ImGuiNET/ImGui.NET#404]
Whenever I want to try get a simple project started quickly I end up either going around in circles trying to get it working, 
or copy pasta/cargo culting in a whole bunch of extraneous stuff just to get something working, and it just looks messy and bloated. 

Something along the lines of:
a list of nuget packages 
a chunk of code or Program.cs to set it up. 

with a small section of extra stuff you might want. (i.e. implot)

I realize there's a couple of ways to do this depending on what render you're wanting to use, but that shouldn't complicate stuff too much I don't think.

**Comment by zaafar** [comment]
https://github.com/ImGuiNET/ImGui.NET/tree/master/src/ImGui.NET.SampleProgram is not enough?

**Comment by se5a** [comment]
not really, 
there's six other projects in that, which is fine if you're wanting to futz around with imgui.net project itself, but not so great if you just want to get something bare bones working for a project.

**Comment by se5a** [comment]
Why can't we have instructions for doing it ourselfs?
I want to start from zero and add stuff, not start from bloat and remove stuff.

**Comment by zaafar** [comment]
> I want to start from zero and add stuff, not start from bloat and remove stuff.

Define zero? I think your definition of zero is very different from (users of this library) definition of zero.

**Comment by se5a** [comment]
I think my op and following comments make it clear what I'm asking for.

**Comment by zaafar** [comment]
> I think my op and following comments make it clear what I'm asking for.

Actually it does not, what backend you want in that wiki? Imgui doesn't work without a backend, so what does ZERO mean over here? Does it mean with or without backend?

FYI: Following are the list of backends imgui supports.
https://github.com/ocornut/imgui/tree/master/backends

Imgui doesn't work without keyboard, mouse, controller input system, what input system you want in that wiki? There are 1000s of configuration that be different when using imgui so which configuration should be considered ZERO and included in the wiki?

**Comment by se5a** [comment]
I really feel like I'm being *deliberately* misunderstood and dismissed here. 

I challenge you to open visual studio/code/rider or whatever your editor of choice is, create a new project and figure out exactly what nuget packages you need, drop in the code from the SampleProgram and get it working.  

If I understand correctly, veldrid libraries handle most of the backend stuff, use those. and/or use openGL just for the cross platform ability.

**Comment by bootzin** [comment]
Se5a, you're not being misunderstood.
You're failing to understand.
Let me try to explain: ImGui is a immediate Gui library. ImGui.Net is a dot net wrapper for that.
Imgui (and therefore imgui.net) has no way of opening windows, nor drawing shapes to the screen. It doesn't know rendering, it simply allows for structured creation of a GUI.
If you want to manage windows, you need a separate library for that. If you wanna render fonts, you need a dedicated graphics library for that. ImGui does not support that by default
It doesn't have sdl2 as a dependency, nor does it has opengl as a dependency. You can use whatever you want
So there is no MVP with ImGui.Net alone. Conversely, there's no required dependencies either.
If you wanna do a barebones project, you absolutely _need_ to understand at least one of the samples. You don't need to use it in your project, but you need to understand what it's doing, how it's doing, and how to adapt it to your own project.
This is not a math library that you can just drop in any project. You need to study and experiment with it.
And, for the record, before I tried anything with ImGui, I was a proficient C# developer. And yes, it was challenging as hell, but what's provided in the samples is more than enough to get you going, all you need is to be willing to learn, and not copy paste everything.
Lastly: a window manager library and a graphics rendering library should be everything you need to get started with ImGui. Check the original repo (in cpp). It's not hard to adapt, and you'll see they'll have a lot of examples, for a plethora of different window managers and graphics libraries. Study, learn, and one day, you'll see we were right, afterall

**Comment by DarthPointer** [comment]
I agree that this project really needs an introduction on what to add to make the library work.

Requiring the users to find themselves out what to include is just unreasonable. End-users of UI frameworks are those who design user UI apps, not necessarily familiar with all the layers of app rendering pipeline. Having these template projects is not enough. They really combine together few things that are unnecessary detail to people trying the framework out for the first time to see if its approach fits them. Reading these on plain github repo view page is not much helpful and downloading the 

I don't say the entire library should be graspable within 10 minutes. If it does some crazy shit under the hood with lots of specific detail for fine tune then no one could understand it in these 10 minutes. But an entry level tutorial to start using the product is crucial. You won't get large userbase if everybody entering will be met with the 10-metre high brickwall of undocumented sample projects. If you want people to enjoy your product you *must* tell them how to get first experience with *the product,* not the poor exception traces. I may not assume the reason for the people behind ImGUI.Net to create and support it. But if they want to make something great and share it with people, pretending the current non-existence of tutorial is fine is plain hypocrisy. The tools and libraries you used to create this wrap had their documentation and abstractions good enough to let you focus on the port, not finding out the ways how to implement CPU and GPU-specific optimizations, hadn't they?

I have no doubt that it took great effort to make the original DearGUI work and this .Net wrap run, but it is embarassing how low grade the entry level experience is when working with it for the first time. Thousands of hours spent are suffering extreme loss of value because no one made a get-and-run page which could take no more than 10 hours to write. So that people could at least see that the library works and they would start learning it before sinking their own hours into selecting lower levels of their app. Which may be completely irrelevant for small applications where UI performance and other lib differences won't be noticeable.

**Comment by bootzin** [comment]
> because no one made a get-and-run page which could take no more than 10 hours to write

Listen, if you're so keen on having this sort of documentation, you are _very_ welcome to contribute to the repo. Take a moment to learn it as it is and write about your findings along the way. Then compile those findings in a document and submit a PR. Everyone would appreciate the initiative.

**Comment by DarthPointer** [comment]
Yeah taking a "moment" to learn means I will need to dedicate a full day or few to first learn, then write a guide and then check that I didn't miss something in the process.

The amount of time it would take anyone new to make such a guide is way more than for someone who has experience with it and has already created projects that use it. I found it out that for me it is too time-demanding and unreasonable to relearn from the available "documentation" to bother with this library. Why should I bother pointing out X if it has already been done by others? This is past for breaking point for me to get the thing working. It makes me ask questions about the creators are being interested in maintaining and supporting the product. About having a chance to get help if I encounter an issue.

I do respect the effort to build a wheel, but it sinks in futility if everyone else is forced to reinvent it anyway.

**Comment by zaafar** [comment]
I am going to lock this issue now, if anyone has some time to do this, feel free to create a PR. This should be a good first issue to solve. :)
